### PR TITLE
repair manager - email alerting refactoring

### PR DESCRIPTION
**Pull request by debbie-alaine**
Migrated some of the alerting logic to RuleAlertHandler to handle caching and reminders, attempting to decouple from any specific rule.

The RuleAlertHandler now includes a ttl-cache window, so that if a node has not been seen for a configured time (currently set to one hour), it will send an alert email. this was an issue in one of the clusters were the prometheus query was not consistent in the nodes (switching from 2 nodes with ecc error, to 3, then back to 2 every couple of minutes). This was causing many emails.

### CODE REVIEW

**Comment by debbie-alaine** [comment]
@Anbang-Hu  @hongzhili  @YinYangOfDao let me know if you have any feedback.

File: src/dev-utils/prometheus_url.py [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
-
-    parser.add_argument("--since", "-s", type=int, default=ago,
-            help="start time for generating report")
-    parser.add_argument("--until", "-u", type=int, default=now,
-            help="end time for generating report")
-
-    args = parser.parse_args()
-
-    main(args)
+#!/usr/bin/env python3
```
Comment (Anbang-Hu) [inline src/dev-utils/prometheus_url.py]: Why is this file showing diff?

File: src/RepairManager/test/test_ecc_rule.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
+import os, sys
```
Comment (Anbang-Hu) [inline src/RepairManager/test/test_ecc_rule.py]: Can you follow #769 to enable tests for repairmanager?

File: src/dev-utils/prometheus_url.py (hunk 1 above)
Comment (debbie-alaine) [reply src/dev-utils/prometheus_url.py]: oops, should not have updated here. though I did update it in common utils in Repair Manager. will fix.

**Pull request merged**